src/helpers/text_helper.ts

```diff
@@ -224,6 +224,40 @@ export function getContextFontSize(font: string): Pixel {
   return Number(font.match(pxRegex)?.[1]);
 }
 
+export function drawSplittedText(
+  context: CanvasRenderingContext2D,
+  text: string,
+  position: PixelPosition
+) {
+  const splitted = splitText(text);
+  const measure = context.measureText(splitted[1]);
+  context.fillText(splitted[1], position.x, position.y);
+  position.y -= measure.actualBoundingBoxAscent + measure.actualBoundingBoxDescent + 2;
+  context.fillText(splitted[0], position.x, position.y);
+}
+
+export function splitText(text: string): string[] {
+  let spaces = "";
+  while (text[0] === " ") {
+    spaces += " ";
+    text = text.slice(1);
+  }
+  const length = Math.floor(text.length);
+  let leftSpace = Math.floor(length / 2),
+    rightSpace = Math.floor(length / 2);
+  while (leftSpace > 0 && text[leftSpace] !== " ") {
+    leftSpace--;
+  }
+  while (rightSpace < text.length && text[rightSpace] !== " ") {
+    rightSpace++;
+  }
```

Comment:
You cen definetly just trim() rather than looping there

Reply:
I'm not sure you understood the loop. Here I'm searching for the first space right and left of the middle to know where to split the line. I don't see how a trim can help me doing that ?